prow/cmd/deck/static/fuzzy-search.js

```diff
@@ -54,62 +54,71 @@ limitations under the License.
      *  1. +3 score for the matching that occurs near the beginning of the string.
      *  2. +5 score for the matching that is not an alphabetical character.
      *  3. +3 score for the matching that the string character is upper case.
-     *  4. +10 score for the matching that matches the uppercase which is just before a 
+     *  4. +10 score for the matching that matches the uppercase which is just before a
      *  separator.
      * @param {number} i
      * @param {string} str
      * @return {number}
-     */ 
+     */
     this.calcScore = function(i, str) {
       var score = 0;
-      var isNotAlphabetical = function (c) {
-        return (c < 65 || (c > 90 && c < 97) || c > 122);
+      var isAlphabetical = function (c) {
+        return (c > 64 && c < 91) || (c > 96 && c < 123);
       };
       // Bonus if the matching is near the start of the string
       if (i < 3) {
         score += 3;
-      } 
+      }
       // Bonus if the matching is not a alphabetical character
-      if (isNotAlphabetical(str.charCodeAt(i))) {
+      if (!isAlphabetical(str.charCodeAt(i))) {
         score += 5;
       }
       // Bonus if the matching is an UpperCase character
       if (str[i].toUpperCase() === str[i]) {
         score += 3;
       }
-      
+
       // Bonus if matching after a separator
-      var separatorBehind = (i === 0 || isNotAlphabetical(str.charCodeAt(i - 1)));
-      if (separatorBehind) {
+      var separatorBehind = (i === 0 || !isAlphabetical(str.charCodeAt(i - 1)));
+      if (separatorBehind && isAlphabetical(str.charCodeAt(i))) {
         score += 10;
         score += (str[i].toUpperCase() === str[i] ? 5 : 0);
       }
       return score;
     };
-    
+
     /**
      * Get maximum score that a string can get against the pattern.
      * @param {string} pttn
      * @param {string} str
      * @return {number}
      */
     this.getMaxScore = function(pttn, str) {
+      if (pttn === str) {
+        return 1000000000;
```

Comment:
this value seems somewhat arbitrary.

Reply:
all score values are arbitrary

Reply:
They're not, otherwise they'd be useless :-)
A magical value like this should have some comment with a rationale, or be less magical like `Number.MAX_SAFE_INTEGER`.

Reply:
I made the changes which made the score more reasonable.